js/index.js

```diff
@@ -1,7 +1,144 @@
+let frames = 0;
+
 window.onload = () => {
-  document.getElementById('start-button').onclick = () => {
+  document.getElementById("start-button").onclick = () => {
     startGame();
-  };
+  }
 
-  function startGame() {}
+  function startGame() {
+    setInterval(updateGame, 20);
+  }
 };
+
+const canvas = document.getElementById("canvas");
+const context = canvas.getContext("2d");
+
+class Game {
+  constructor(x, y, width, height) {
+    this.img = new Image();
+    this.img.src = "./images/road.png";
+    this.width = width;
+    this.height = height;
+    this.x = x;
+    this.y = y;
+    this.score = 0;
+  }
+
+  drawRoad() {
+    // drawImage(img, x, y, width, height)
+    context.drawImage(this.img, this.x, this.y, this.width, this.height);
+  }
+
+  clearRoad() {
+    context.clearRect(this.x, this.y, this.width, this.height);
+  }
+}
+
+class Car {
+  constructor(x, y, width, height) {
+    this.img = new Image();
+    this.img.src = "./images/car.png";
+    this.width = width;
+    this.height = height;
+    this.x = x;
+    this.y = y;
+    this.speed = 5;
+  }
+
+  drawCar() {
+    // this.moveCar(direction)
+    context.drawImage(this.img, this.x, this.y, this.width, this.height);
+  }
+
+  moveLeft() {
+    this.x -= this.speed;
+  }
+
+  moveRight() {
+    this.x += this.speed;
+  }s
+
+  clearCar() {
+    context.clearRect(this.x, this.y, this.width, this.height);
+  }
+
+  detectWalls() {
+    if (this.x < 50) {
+      this.x = 50;
+    } else if (this.x + this.width > 450) {
+      this.x = 450 - this.width;
+    }
+  }
+
+  colidingObstacles() {
+    if (this.x + this.width === obstacles.y) {
+      return true;
+    } else if (this.x === obstacles.y + obstacles.width) {
+      return true;
+    } else if (this.y + this.width === obstacles.y) {
+      return true;
+    } else if (this.y === obstacles.y + obstacles.width) {
+      return true;
+    } else {
+      return true;
+    }
+  }
+}
+
+class Obstacles {
+  constructor(x, width) {
+    // this.img = new Image();
+    // this.img.src = "./images/car.png";
```

Comment:
Since you are using `fillRect` to draw your obstacle these two commented lines can be removed